x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -305,6 +306,12 @@ private void createApiKeyAndIndexIt(
         final SecureString apiKey = UUIDs.randomBase64UUIDSecureString();
         final Version version = clusterService.state().nodes().getMinNodeVersion();
 
+        if (TcpTransport.isUntrustedRemoteClusterEnabled()
```

Comment:
This is a temporary stop gap. As [mentioned](https://github.com/elastic/elasticsearch/pull/90614#issuecomment-1273734312) I'm planning to add support for API keys in a follow up PR. It's a fairly big change.